internal/pkg/cli/svc_init.go

```diff
@@ -222,18 +220,38 @@ func (o *initSvcOpts) Validate() error {
 	return nil
 }
 
-// Ask prompts for fields that are required but not passed in.
+// Ask prompts for and validates any required flags.
 func (o *initSvcOpts) Ask() error {
-	if err := o.askSvcName(); err != nil {
-		return err
+	if o.wkldType != "" {
+		if err := validateSvcType(o.wkldType); err != nil {
+			return err
+		}
+	} else {
+		if err := o.askSvcType(); err != nil {
+			return err
+		}
+	}
+	if o.name != "" {
+		if err := validateSvcName(o.name, o.wkldType); err != nil {
+			return err
+		}
+		if err := o.validateDuplicateSvc(); err != nil {
+			return err
+		}
+	} else {
+		if err := o.askSvcName(); err != nil {
```

Comment:
Looking at the code this function sounds more like a `askAndValidateSvcName` to me...

We decided on the `if o.name != "" { // validate } else { //ask}` pattern because the questions being asked are selection prompts, i.e. they don't need to be further validated.

But this is an input prompt, and we still need to validate the input after prompting. Do you think we should instead do
```
if o.name != {
  // ask
}
// validate
```
?

Reply:
tbh this is less ideal as if they don't have flag specified we did `validateSvcName` twice.

Reply:
looks better tho so i made the change